### ISSUE TITLE
Bug: A database statement execution error occurred: NOT NULL constraint failed: item.type

### ISSUE THREAD

**Issue by DieBe** [abraunegg/onedrive#2938]
### Describe the bug

When performing a sync, for some files and directories, the OneDrive API doesn't return the standard elements that it is supposed to, according to the documentation. This leads to the following error messages:
```A database statement execution error occurred: NOT NULL constraint failed: item.type``` (occuring much more often)
or 
```A database statement execution error occurred: FOREIGN KEY constraint failed``` (rarer)

### Operating System Details

```shell
Linux xpsbenjamin 6.11.0-9-generic #9-Ubuntu SMP PREEMPT_DYNAMIC Mon Oct 14 13:19:59 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux

No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 24.10
Release:        24.10
Codename:       oracular
```

### Client Installation Method

From Source

### OneDrive Account Type

Personal

### What is your OneDrive Application Version

onedrive v2.5.2-25-g2267a23

### What is your OneDrive Application Configuration

```shell
--------------- Application Runtime Configuration ---------------
Application version                          = onedrive v2.5.2-13-g967327c
Compiled with                                = DMD 2109
User Application Config path                 = /home/benjamin/.config/onedrive
System Application Config path               = /etc/onedrive
Applicable Application 'config' location     = /home/benjamin/.config/onedrive/config
Configuration file found in config location  = true - using 'config' file values to override application defaults
Applicable 'sync_list' location              = /home/benjamin/.config/onedrive/sync_list
Applicable 'items.sqlite3' location          = /home/benjamin/.config/onedrive/items.sqlite3
Config option 'drive_id'                     = 
Config option 'sync_dir'                     = ~/OneDrive
Config option 'enable_logging'               = false
Config option 'log_dir'                      = /var/log/onedrive
Config option 'disable_notifications'        = false
Config option 'skip_dir'                     = .Trash-1000|Application\ Data|Dokumente
Config option 'skip_dir_strict_match'        = true
Config option 'skip_file'                    = ~*|.~*|*.tmp|*.swp|*.partial
Config option 'skip_dotfiles'                = false
Config option 'skip_symlinks'                = false
Config option 'monitor_interval'             = 300
Config option 'monitor_log_frequency'        = 12
Config option 'monitor_fullscan_frequency'   = 12
Config option 'read_only_auth_scope'         = false
Config option 'dry_run'                      = false
Config option 'upload_only'                  = false
Config option 'download_only'                = false
Config option 'local_first'                  = false
Config option 'check_nosync'                 = false
Config option 'check_nomount'                = false
Config option 'resync'                       = true
Config option 'resync_auth'                  = false
Config option 'cleanup_local_files'          = false
Config option 'classify_as_big_delete'       = 1000
Config option 'disable_upload_validation'    = false
Config option 'disable_download_validation'  = false
Config option 'bypass_data_preservation'     = false
Config option 'no_remote_delete'             = false
Config option 'remove_source_files'          = false
Config option 'sync_dir_permissions'         = 700
Config option 'sync_file_permissions'        = 600
Config option 'space_reservation'            = 52428800
Config option 'application_id'               = d50ca740-c83f-4d1b-b616-12c519384f0c
Config option 'azure_ad_endpoint'            = 
Config option 'azure_tenant_id'              = 
Config option 'user_agent'                   = ISV|abraunegg|OneDrive Client for Linux/v2.5.2-13-g967327c
Config option 'force_http_11'                = false
Config option 'debug_https'                  = false
Config option 'rate_limit'                   = 0
Config option 'operation_timeout'            = 3600
Config option 'dns_timeout'                  = 60
Config option 'connect_timeout'              = 10
Config option 'data_timeout'                 = 60
Config option 'ip_protocol_version'          = 0
Config option 'threads'                      = 14
Compile time option --enable-notifications   = false

Selective sync 'sync_list' configured        = true
sync_list config option 'sync_root_files'    = false
sync_list contents:



!/Bilder/*
!/Desktop/*
!lib/
!cache/

/temp/*
/Dokumente/*
/Backup/*
/thunderbird/*
/Code/*

Config option 'sync_business_shared_items'   = false

Config option 'webhook_enabled'              = false

--------------------DEVELOPER_OPTIONS----------------------------
Config option 'force_children_scan'          = false

-----------------------------------------------------------------
```

### What is your 'curl' version

```shell
curl 8.9.1 (x86_64-pc-linux-gnu) libcurl/8.9.1 OpenSSL/3.3.1 zlib/1.3.1 brotli/1.1.0 zstd/1.5.6 libidn2/2.3.7 libpsl/0.21.2 libssh2/1.11.0 nghttp2/1.62.1 librtmp/2.3 OpenLDAP/2.6.8
Release-Date: 2024-07-31, security patched: 8.9.1-2ubuntu2
Protocols: dict file ftp ftps gopher gophers http https imap imaps ipfs ipns ldap ldaps mqtt pop3 pop3s rtmp rtsp scp sftp smb smbs smtp smtps telnet tftp ws wss
Features: alt-svc AsynchDNS brotli GSS-API HSTS HTTP2 HTTPS-proxy IDN IPv6 Kerberos Largefile libz NTLM PSL SPNEGO SSL threadsafe TLS-SRP UnixSockets zstd
```

### Where is your 'sync_dir' located

Local

### What are all your system 'mount points'

```shell
sysfs on /sys type sysfs (rw,nosuid,nodev,noexec,relatime)
proc on /proc type proc (rw,nosuid,nodev,noexec,relatime)
udev on /dev type devtmpfs (rw,nosuid,relatime,size=16176468k,nr_inodes=4044117,mode=755,inode64)
devpts on /dev/pts type devpts (rw,nosuid,noexec,relatime,gid=5,mode=620,ptmxmode=000)
tmpfs on /run type tmpfs (rw,nosuid,nodev,noexec,relatime,size=3243316k,mode=755,inode64)
/dev/nvme0n1p2 on / type ext4 (rw,relatime)
securityfs on /sys/kernel/security type securityfs (rw,nosuid,nodev,noexec,relatime)
tmpfs on /dev/shm type tmpfs (rw,nosuid,nodev,inode64)
cgroup2 on /sys/fs/cgroup type cgroup2 (rw,nosuid,nodev,noexec,relatime,nsdelegate,memory_recursiveprot)
pstore on /sys/fs/pstore type pstore (rw,nosuid,nodev,noexec,relatime)
efivarfs on /sys/firmware/efi/efivars type efivarfs (rw,nosuid,nodev,noexec,relatime)
bpf on /sys/fs/bpf type bpf (rw,nosuid,nodev,noexec,relatime,mode=700)
systemd-1 on /proc/sys/fs/binfmt_misc type autofs (rw,relatime,fd=34,pgrp=1,timeout=0,minproto=5,maxproto=5,direct,pipe_ino=9269)
hugetlbfs on /dev/hugepages type hugetlbfs (rw,nosuid,nodev,relatime,pagesize=2M)
mqueue on /dev/mqueue type mqueue (rw,nosuid,nodev,noexec,relatime)
tmpfs on /run/lock type tmpfs (rw,nosuid,nodev,noexec,relatime,size=5120k,inode64)
none on /sys/kernel/debug type debugfs (rw,nosuid,nodev,noexec,relatime)
tracefs on /sys/kernel/tracing type tracefs (rw,nosuid,nodev,noexec,relatime)
tmpfs on /run/credentials/systemd-journald.service type tmpfs (ro,nosuid,nodev,noexec,relatime,nosymfollow,size=1024k,nr_inodes=1024,mode=700,inode64,noswap)
tmpfs on /run/credentials/systemd-udev-load-credentials.service type tmpfs (ro,nosuid,nodev,noexec,relatime,nosymfollow,size=1024k,nr_inodes=1024,mode=700,inode64,noswap)
configfs on /sys/kernel/config type configfs (rw,nosuid,nodev,noexec,relatime)
fusectl on /sys/fs/fuse/connections type fusectl (rw,nosuid,nodev,noexec,relatime)
tmpfs on /run/credentials/systemd-tmpfiles-setup-dev-early.service type tmpfs (ro,nosuid,nodev,noexec,relatime,nosymfollow,size=1024k,nr_inodes=1024,mode=700,inode64,noswap)
tmpfs on /run/credentials/systemd-tmpfiles-setup-dev.service type tmpfs (ro,nosuid,nodev,noexec,relatime,nosymfollow,size=1024k,nr_inodes=1024,mode=700,inode64,noswap)
tmpfs on /run/credentials/systemd-sysctl.service type tmpfs (ro,nosuid,nodev,noexec,relatime,nosymfollow,size=1024k,nr_inodes=1024,mode=700,inode64,noswap)
/var/lib/snapd/snaps/bare_5.snap on /snap/bare/5 type squashfs (ro,nodev,relatime,errors=continue,threads=single,x-gdu.hide,x-gvfs-hide)
/var/lib/snapd/snaps/chromium_2980.snap on /snap/chromium/2980 type squashfs (ro,nodev,relatime,errors=continue,threads=single,x-gdu.hide,x-gvfs-hide)
/var/lib/snapd/snaps/clion_292.snap on /snap/clion/292 type squashfs (ro,nodev,relatime,errors=continue,threads=single,x-gdu.hide,x-gvfs-hide)
/var/lib/snapd/snaps/core18_2829.snap on /snap/core18/2829 type squashfs (ro,nodev,relatime,errors=continue,threads=single,x-gdu.hide,x-gvfs-hide)
/var/lib/snapd/snaps/clion_296.snap on /snap/clion/296 type squashfs (ro,nodev,relatime,errors=continue,threads=single,x-gdu.hide,x-gvfs-hide)
/var/lib/snapd/snaps/core18_2846.snap on /snap/core18/2846 type squashfs (ro,nodev,relatime,errors=continue,threads=single,x-gdu.hide,x-gvfs-hide)
/var/lib/snapd/snaps/core20_2379.snap on /snap/core20/2379 type squashfs (ro,nodev,relatime,errors=continue,threads=single,x-gdu.hide,x-gvfs-hide)
/var/lib/snapd/snaps/core20_2434.snap on /snap/core20/2434 type squashfs (ro,nodev,relatime,errors=continue,threads=single,x-gdu.hide,x-gvfs-hide)
/var/lib/snapd/snaps/core22_1663.snap on /snap/core22/1663 type squashfs (ro,nodev,relatime,errors=continue,threads=single,x-gdu.hide,x-gvfs-hide)
/var/lib/snapd/snaps/cups_1058.snap on /snap/cups/1058 type squashfs (ro,nodev,relatime,errors=continue,threads=single,x-gdu.hide,x-gvfs-hide)
/var/lib/snapd/snaps/core22_1621.snap on /snap/core22/1621 type squashfs (ro,nodev,relatime,errors=continue,threads=single,x-gdu.hide,x-gvfs-hide)
/var/lib/snapd/snaps/cups_1067.snap on /snap/cups/1067 type squashfs (ro,nodev,relatime,errors=continue,threads=single,x-gdu.hide,x-gvfs-hide)
/var/lib/snapd/snaps/firmware-updater_127.snap on /snap/firmware-updater/127 type squashfs (ro,nodev,relatime,errors=continue,threads=single,x-gdu.hide,x-gvfs-hide)
/var/lib/snapd/snaps/gnome-3-28-1804_198.snap on /snap/gnome-3-28-1804/198 type squashfs (ro,nodev,relatime,errors=continue,threads=single,x-gdu.hide,x-gvfs-hide)
/var/lib/snapd/snaps/gnome-42-2204_176.snap on /snap/gnome-42-2204/176 type squashfs (ro,nodev,relatime,errors=continue,threads=single,x-gdu.hide,x-gvfs-hide)
/var/lib/snapd/snaps/gnome-3-38-2004_143.snap on /snap/gnome-3-38-2004/143 type squashfs (ro,nodev,relatime,errors=continue,threads=single,x-gdu.hide,x-gvfs-hide)
/var/lib/snapd/snaps/gtk-common-themes_1535.snap on /snap/gtk-common-themes/1535 type squashfs (ro,nodev,relatime,errors=continue,threads=single,x-gdu.hide,x-gvfs-hide)
/var/lib/snapd/snaps/gtk-common-themes_1536.snap on /snap/gtk-common-themes/1536 type squashfs (ro,nodev,relatime,errors=continue,threads=single,x-gdu.hide,x-gvfs-hide)
/var/lib/snapd/snaps/icon-theme-breeze_4.snap on /snap/icon-theme-breeze/4 type squashfs (ro,nodev,relatime,errors=continue,threads=single,x-gdu.hide,x-gvfs-hide)
/var/lib/snapd/snaps/nvim_2938.snap on /snap/nvim/2938 type squashfs (ro,nodev,relatime,errors=continue,threads=single,x-gdu.hide,x-gvfs-hide)
/var/lib/snapd/snaps/snapd_21465.snap on /snap/snapd/21465 type squashfs (ro,nodev,relatime,errors=continue,threads=single,x-gdu.hide,x-gvfs-hide)
/var/lib/snapd/snaps/r-quick-share_x1.snap on /snap/r-quick-share/x1 type squashfs (ro,nodev,relatime,errors=continue,threads=single,x-gdu.hide,x-gvfs-hide)
/var/lib/snapd/snaps/snapd_21759.snap on /snap/snapd/21759 type squashfs (ro,nodev,relatime,errors=continue,threads=single,x-gdu.hide,x-gvfs-hide)
/var/lib/snapd/snaps/thunderbird_526.snap on /snap/thunderbird/526 type squashfs (ro,nodev,relatime,errors=continue,threads=single,x-gdu.hide,x-gvfs-hide)
/var/lib/snapd/snaps/thunderbird_532.snap on /snap/thunderbird/532 type squashfs (ro,nodev,relatime,errors=continue,threads=single,x-gdu.hide,x-gvfs-hide)
/var/lib/snapd/snaps/whatsapp-for-linux_59.snap on /snap/whatsapp-for-linux/59 type squashfs (ro,nodev,relatime,errors=continue,threads=single,x-gdu.hide,x-gvfs-hide)
/var/lib/snapd/snaps/webkitgtk-6-gnome-2204_31.snap on /snap/webkitgtk-6-gnome-2204/31 type squashfs (ro,nodev,relatime,errors=continue,threads=single,x-gdu.hide,x-gvfs-hide)
tmpfs on /tmp type tmpfs (rw,noatime,inode64)
/dev/nvme0n1p1 on /boot/efi type vfat (rw,relatime,fmask=0022,dmask=0022,codepage=437,iocharset=iso8859-1,shortname=mixed,errors=remount-ro)
tmpfs on /run/credentials/systemd-tmpfiles-setup.service type tmpfs (ro,nosuid,nodev,noexec,relatime,nosymfollow,size=1024k,nr_inodes=1024,mode=700,inode64,noswap)
tmpfs on /run/credentials/systemd-resolved.service type tmpfs (ro,nosuid,nodev,noexec,relatime,nosymfollow,size=1024k,nr_inodes=1024,mode=700,inode64,noswap)
binfmt_misc on /proc/sys/fs/binfmt_misc type binfmt_misc (rw,nosuid,nodev,noexec,relatime)
tmpfs on /run/snapd/ns type tmpfs (rw,nosuid,nodev,noexec,relatime,size=3243316k,mode=755,inode64)
nsfs on /run/snapd/ns/cups.mnt type nsfs (rw)
tmpfs on /run/user/1000 type tmpfs (rw,nosuid,nodev,relatime,size=3243312k,nr_inodes=810828,mode=700,uid=1000,gid=1000,inode64)
portal on /run/user/1000/doc type fuse.portal (rw,nosuid,nodev,relatime,user_id=1000,group_id=1000)
/var/lib/snapd/snaps/chromium_2985.snap on /snap/chromium/2985 type squashfs (ro,nodev,relatime,errors=continue,threads=single,x-gdu.hide,x-gvfs-hide)
nsfs on /run/snapd/ns/chromium.mnt type nsfs (rw)
tracefs on /sys/kernel/debug/tracing type tracefs (rw,nosuid,nodev,noexec,relatime)
nsfs on /run/snapd/ns/firmware-updater.mnt type nsfs (rw)
/var/lib/snapd/snaps/nvim_2962.snap on /snap/nvim/2962 type squashfs (ro,nodev,relatime,errors=continue,threads=single,x-gdu.hide,x-gvfs-hide)
nsfs on /run/snapd/ns/whatsapp-for-linux.mnt type nsfs (rw)
/var/lib/snapd/snaps/icon-theme-breeze_5.snap on /snap/icon-theme-breeze/5 type squashfs (ro,nodev,relatime,errors=continue,threads=single,x-gdu.hide,x-gvfs-hide)
```

### What are all your local file system partition types

```shell
NAME        FSTYPE   FSVER LABEL        UUID                                 FSAVAIL FSUSE% MOUNTPOINTS
loop0       squashfs 4.0                                                           0   100% /snap/bare/5
loop1       squashfs 4.0                                                           0   100% /snap/nvim/2962
loop2       squashfs 4.0                                                           0   100% /snap/chromium/2980
loop3       squashfs 4.0                                                           0   100% /snap/clion/292
loop4       squashfs 4.0                                                           0   100% /snap/clion/296
loop5       squashfs 4.0                                                           0   100% /snap/core18/2829
loop6       squashfs 4.0                                                           0   100% /snap/core20/2379
loop7       squashfs 4.0                                                           0   100% /snap/core18/2846
loop8       squashfs 4.0                                                           0   100% /snap/core20/2434
loop9       squashfs 4.0                                                           0   100% /snap/core22/1663
loop10      squashfs 4.0                                                           0   100% /snap/cups/1058
loop11      squashfs 4.0                                                           0   100% /snap/core22/1621
loop12      squashfs 4.0                                                           0   100% /snap/cups/1067
loop13      squashfs 4.0                                                                    
loop14      squashfs 4.0                                                           0   100% /snap/icon-theme-breeze/5
loop15      squashfs 4.0                                                           0   100% /snap/gnome-3-28-1804/198
loop16      squashfs 4.0                                                           0   100% /snap/firmware-updater/127
loop17      squashfs 4.0                                                           0   100% /snap/gnome-42-2204/176
loop18      squashfs 4.0                                                           0   100% /snap/gnome-3-38-2004/143
loop19      squashfs 4.0                                                           0   100% /snap/gtk-common-themes/1535
loop20      squashfs 4.0                                                           0   100% /snap/gtk-common-themes/1536
loop21      squashfs 4.0                                                           0   100% /snap/icon-theme-breeze/4
loop23      squashfs 4.0                                                           0   100% /snap/nvim/2938
loop24      squashfs 4.0                                                           0   100% /snap/r-quick-share/x1
loop25      squashfs 4.0                                                           0   100% /snap/snapd/21465
loop26      squashfs 4.0                                                           0   100% /snap/snapd/21759
loop27      squashfs 4.0                                                           0   100% /snap/thunderbird/526
loop28      squashfs 4.0                                                           0   100% /snap/thunderbird/532
loop29      squashfs 4.0                                                           0   100% /snap/whatsapp-for-linux/59
loop30      squashfs 4.0                                                           0   100% /snap/webkitgtk-6-gnome-2204/31
loop31      squashfs 4.0                                                           0   100% /snap/chromium/2985
nvme0n1                                                                                     
├─nvme0n1p1 vfat     FAT32              28E1-1051                             255.8M    15% /boot/efi
└─nvme0n1p2 ext4     1.0   kubuntu_2404 b64b3db1-0cd0-42be-80a7-a88e7fd1b98f  510.1G    39% /
```

### How do you use 'onedrive'

OneDrive is also used from a Windows PC and two Android devices, occasionally from the Web UI

### Steps to reproduce the behaviour

Perfom any ```--sync```or ```--monitor```operation, no matter if with or without ```--resync```. A complete, clean reinstall also doesn't resolve it.

### Complete Verbose Log Output

```shell
(Skipping lots of successful API calls)

Using IPv4 and IPv6 (if configured) for all network operations
Attempting to contact Microsoft OneDrive Login Service
Successfully reached Microsoft OneDrive Login Service
Checking Application Version ...
Attempting to initialise the OneDrive API ...
Configuring Global Azure AD Endpoints
The OneDrive API was initialised successfully
Opening the item database ...
Application Version:  onedrive v2.5.2-25-g2267a23
Account Type:         personal
Default Drive ID:     cbbc790e554348af
Default Root ID:      CBBC790E554348AF!101
Remaining Free Space: 683.60 GB (734009910886 bytes)
Sync Engine Initialised with new Onedrive API instance
All application operations will be performed in the configured local 'sync_dir' directory: /home/benjamin/OneDrive
Fetching /delta response from the OneDrive API for Drive ID: cbbc790e554348af
Processing API Response Bundle: 1 - Quantity of 'changes|items' in this bundle to process: 224
Skipping path - excluded by sync_list config: ./Musik
Including path - included by sync_list config: ./temp
Skipping path - excluded by sync_list config: ./Bilder
Including path - included by sync_list config: ./Backup
Skipping path - excluded by sync_list config: ./Geteilte Dateien
Including path - included by sync_list config: ./Noten
Including path - included by sync_list config: ./Dokumente

...

Including file - included by sync_list config: Dokumente/ETH/notes.one
Parental Path structure needs to be created to support included file: Dokumente/ETH

A database statement execution error occurred: NOT NULL constraint failed: item.type

...

Including file - included by sync_list config: Dokumente/ETH/Open Notebook.onetoc2
Parental Path structure needs to be created to support included file: Dokumente/ETH

A database statement execution error occurred: NOT NULL constraint failed: item.type

Please restart the application with --resync to potentially fix any local database issues.
Skipping path - excluded by sync_list config: Dokumente/Arduino/libraries
Including file - included by sync_list config: Dokumente/ETH/0429.one
Parental Path structure needs to be created to support included file: Dokumente/ETH

A database statement execution error occurred: NOT NULL constraint failed: item.type

Please restart the application with --resync to potentially fix any local database issues.
Including file - included by sync_list config: Dokumente/Notizbuch von Benjamin/Quick Notes.one
Parental Path structure needs to be created to support included file: Dokumente/Notizbuch von Benjamin

A database statement execution error occurred: NOT NULL constraint failed: item.type

Please restart the application with --resync to potentially fix any local database issues.

...

Including file - included by sync_list config: Dokumente/ETH/Open Notebook.onetoc2
Parental Path structure needs to be created to support included file: Dokumente/ETH

A database statement execution error occurred: NOT NULL constraint failed: item.type

Please restart the application with --resync to potentially fix any local database issues.
Skipping path - excluded by sync_list config: Dokumente/Arduino/libraries
Including file - included by sync_list config: Dokumente/ETH/0429.one
Parental Path structure needs to be created to support included file: Dokumente/ETH

A database statement execution error occurred: NOT NULL constraint failed: item.type

Please restart the application with --resync to potentially fix any local database issues.
Including file - included by sync_list config: Dokumente/Notizbuch von Benjamin/Quick Notes.one
Parental Path structure needs to be created to support included file: Dokumente/Notizbuch von Benjamin

A database statement execution error occurred: NOT NULL constraint failed: item.type

Please restart the application with --resync to potentially fix any local database issues.

...
```

**Comment by abraunegg** [comment]
@DieBe 
Please can you test PR #2939 to resolve your issue. 

First install all the require platform dependencies to build the client on your respective platforms. Please read https://github.com/abraunegg/onedrive/blob/master/docs/install.md#building-from-source---high-level-requirements and then follow correctly for your platform.

Once this is done, to clone the PR to resolve your issue, you can use a script like the following:
```
#!/bin/bash

PR=2939

rm -rf ./onedrive-pr${PR}
git clone https://github.com/abraunegg/onedrive.git onedrive-pr${PR}
cd onedrive-pr${PR}
git fetch origin pull/${PR}/head:pr${PR}
git checkout pr${PR}

# Configure and Build
./configure --enable-debug --enable-notifications; make clean; make;
./onedrive --version
```

This script will create a local folder called `onedrive-pr2939` with the PR version.

To run the PR, you need to run the client from the PR build directory:
```
./onedrive <any other options needed>
```
To install the PR, you will need to perform sudo make install to install the PR version to your system.

When running the PR, your version should be: `onedrive v2.5.2-26-gdb66e7b` or greater.

### Example:
```
alex@onedrive-client-dev onedrive-pr2939]$ ./onedrive  -s -v --resync --resync-auth 
Reading configuration file: /home/alex/.config/onedrive/config
Configuration file successfully loaded
Using 'user' configuration path for application config and state data: /home/alex/.config/onedrive
Deleting the saved application sync status ...
Using IPv4 and IPv6 (if configured) for all network operations
Attempting to contact Microsoft OneDrive Login Service
Successfully reached Microsoft OneDrive Login Service
Checking Application Version ...
Attempting to initialise the OneDrive API ...
Configuring Global Azure AD Endpoints
The OneDrive API was initialised successfully
Opening the item database ...
Application Version:  onedrive v2.5.2-26-gdb66e7b
Account Type:         personal
Default Drive ID:     66d53be8a5056eca
Default Root ID:      66D53BE8A5056ECA!101
Remaining Free Space: 4.70 GB (5046586572 bytes)
Sync Engine Initialised with new Onedrive API instance
All application operations will be performed in the configured local 'sync_dir' directory: /home/alex/OneDrive
Fetching /delta response from the OneDrive API for Drive ID: 66d53be8a5056eca
Processing API Response Bundle: 1 - Quantity of 'changes|items' in this bundle to process: 172
Skipping path - The Microsoft OneNote Notebook '/Documents/ETH' is not supported by this client
Skipping path - The Microsoft OneNote Notebook File '/Documents/ETH/Open Notebook.onetoc2' is not supported by this client
Skipping path - The Microsoft OneNote Notebook File '/Documents/ETH/Untitled Section.one' is not supported by this client
Processing API Response Bundle: 2 - Quantity of 'changes|items' in this bundle to process: 1
Finished processing /delta JSON response from the OneDrive API
Processing 164 applicable changes and items received from Microsoft OneDrive
............
```


This PR works around the OneDrive API not providing all data for the OneNote items.

**Comment by DieBe** [comment]
I tested the pull request with ```./onedrive --sync```, no errors appeared, everything worked as expected. I couldn't find the OneNote notebooks, so I guess they didn't get downloaded. (Which, at least for me, is no problem at all)

Let me know if you need any specific logs or tests of the pull request.

**Comment by abraunegg** [comment]
Thanks for the confirmation - so at least the original issue you faced is fixed by this PR

**Comment by abraunegg** [comment]
Closing issue as fix is merged into 'master'

**Comment by abraunegg** [comment]
This issue has been automatically locked since there has not been any recent activity after it was closed. Please open a new issue for related bugs.
